packages/js/src/plugins/auctionHouseModule/operations/withdrawFromBuyerAccount.ts

```diff
@@ -0,0 +1,227 @@
+  useOperation<WithdrawOperation>(Key);
+
+/**
+ * @group Operations
+ * @category Types
+ */
+export type WithdrawOperation = Operation<
+  typeof Key,
+  WithdrawInput,
+  WithdrawOutput
+>;
+
+/**
+ * @group Operations
+ * @category Inputs
+ */
+export type WithdrawInput = {
+  /** The Auction House from which escrow buyer withdraws funds. */
+  auctionHouse: Pick<
+    AuctionHouse,
+    | 'address'
+    | 'authorityAddress'
+    | 'hasAuctioneer'
+    | 'treasuryMint'
+    | 'feeAccountAddress'
+  >;
+
+  /**
+   * The buyer who withdraws funds.
+   * This expects a Signer.
+   *
+   * @defaultValue `metaplex.identity()`
+   */
+  buyer?: Signer;
+
+  /**
+   * The Authority key.
+   * It is required when the buyer is not a signer.
+   *
+   * @defaultValue Defaults to not being used.
+   */
+  authority?: Signer;
+
+  /**
+   * The Auctioneer authority key.
+   * It is required when Auction House has Auctioneer enabled.
+   *
+   * @defaultValue No default value.
+   */
+  auctioneerAuthority?: Signer;
+
+  /**
+   * Amount of funds to withdraw.
+   * This can either be in SOL or in the SPL token used by the Auction House as a currency.
+   */
+  amount: SolAmount | SplTokenAmount;
+
+  /** A set of options to configure how the transaction is sent and confirmed. */
+  confirmOptions?: ConfirmOptions;
+};
+
+/**
+ * @group Operations
+ * @category Outputs
+ */
+export type WithdrawOutput = {
+  /** The blockchain response from sending and confirming the transaction. */
+  response: SendAndConfirmTransactionResponse;
+};
+
+/**
+ * @group Operations
+ * @category Handlers
+ */
+export const withdrawOperationHandler: OperationHandler<WithdrawOperation> = {
+  handle: async (operation: WithdrawOperation, metaplex: Metaplex) =>
+    withdrawFromBuyerAccountBuilder(operation.input, metaplex).sendAndConfirm(
+      metaplex,
+      operation.input.confirmOptions
+    ),
+};
+
+// -----------------
+// Builder
+// -----------------
+
+/**
+ * @group Transaction Builders
+ * @category Inputs
+ */
+export type WithdrawBuilderParams = Omit<WithdrawInput, 'confirmOptions'> & {
+  instructionKey?: string;
+};
+
+/**
+ * @group Transaction Builders
+ * @category Contexts
+ */
+export type WithdrawBuilderContext = Omit<WithdrawOutput, 'response'>;
+
+/**
+ * Withdraws funds from the user's buyer escrow account to the given auction house.
+ *
+ * ```ts
+ * const transactionBuilder = metaplex
+ *   .auctionHouse()
+ *   .builders()
+ *   .withdrawFromBuyerAccountBuilder({ auctionHouse, buyer, amount });
+ * ```
+ *
+ * @group Transaction Builders
+ * @category Constructors
+ */
+export const withdrawFromBuyerAccountBuilder = (
+  params: WithdrawBuilderParams,
+  metaplex: Metaplex
+): TransactionBuilder<WithdrawBuilderContext> => {
+  const { auctionHouse, auctioneerAuthority, amount } = params;
+
+  if (auctionHouse.hasAuctioneer && !params.auctioneerAuthority) {
+    throw new AuctioneerAuthorityRequiredError();
+  }
+
```

Comment:
I would add a check here.

What do you think? @lorisleiva @antey13 

```
if (!isSigner(seller) && !isSigner(authority)) {
  throw new RequiresSignerError();
}
```

Reply:
Agreed 👌